### PR TITLE
Move ENV, environment, context stuff to 'ember-environment'

### PR DESCRIPTION
**Pull request by krisselden**
Move ENV stuff, environment, and context to 'ember-environment'.

This is a first step of many to cleanup dependencies and dependency on the `Ember` global.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "ember-cli-sauce": "^1.4.2",
     "ember-cli-yuidoc": "0.8.3",
     "ember-publisher": "0.0.7",
-    "emberjs-build": "0.6.4",
+    "emberjs-build": "emberjs/emberjs-build#9c7b35a",
```
Comment (rwjblue) [inline package.json]: Need to update to 0.7.0 now that it is published

Comment (krisselden) [inline package.json]: I updated, it is running tests.

**Comment by rwjblue** [comment]
@krisselden - This LGTM, and is ready for merge. NPM is being a jerk, and having all sorts of errors downloading emberjs-build@0.7.0.

**Comment by krisselden** [comment]
@rwjblue is there an issue with publishing? is it say to merge with that dependency if it isn't downloading?

**Comment by krisselden** [comment]
I can `curl https://registry.npmjs.org/emberjs-build/-/emberjs-build-0.7.0.tgz` successfully I'm not sure what is going on.

**Comment by stefanpenner** [comment]
@krisselden travis might just be hitting a bad replica/mirror, install locally works fine.

**Comment by rwjblue** [comment]
Thank you very much, @krisselden! This is some really great cleanup work!

**Comment by mmun** [comment]
Awesome :+1:

**Pull request merged**